src/js/firstparties/google-search.js

```diff
@@ -21,6 +30,15 @@ function cleanLink(a) {
   // block event listeners on the link
   a.addEventListener("click", function (e) { e.stopImmediatePropagation(); }, true);
   a.addEventListener("mousedown", function (e) { e.stopImmediatePropagation(); }, true);
+
+  // reassign href when in firefox android
+  if (firefox_android) {
+    let href = new URL(a.href).searchParams.get('q');
+    if (!window.isURL(href)) {
+      return;
```

Comment:
Nit: Update the href if `isURL()` matches instead of exiting if it doesn't. Exiting early is nice when it avoids indenting a lot of code, but here we just have a single statement so it's a bit clearer to be more direct.

Reply:
I didn't mean for the URL validity check to be removed. Fixed in efe0159ec5f9613402d920b84d6e2829162497bd and 546bce0f0fe52043ccd218960245f203b4912ec1.